internal/service/networkcontainer/resource_network_container.go

```diff
@@ -211,40 +211,33 @@ func resourceRead(ctx context.Context, d *schema.ResourceData, meta any) diag.Di
 }
 
 func resourceUpdate(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
+	if !d.HasChange("atlas_cidr_block") && !d.HasChange("region_name") && !d.HasChange("region") && !d.HasChange("regions") {
+		return resourceRead(ctx, d, meta)
+	}
+
 	connV2 := meta.(*config.MongoDBClient).AtlasV2
 	ids := conversion.DecodeStateID(d.Id())
 	projectID := ids["project_id"]
 	containerID := ids["container_id"]
-
-	container := new(admin.CloudProviderContainer)
-
-	if d.HasChange("atlas_cidr_block") {
-		atlasCidrBlock := d.Get("atlas_cidr_block").(string)
-		providerName := d.Get("provider_name").(string)
-		container.AtlasCidrBlock = &atlasCidrBlock
-		container.ProviderName = &providerName
-	}
-
-	if d.HasChange("provider_name") {
-		providerName := d.Get("provider_name").(string)
-		container.ProviderName = &providerName
+	regionName, _ := conversion.ValRegion(d.Get("region_name"))
+	region, _ := conversion.ValRegion(d.Get("region"))
+
+	params := &admin.CloudProviderContainer{
```

Comment:
One behaviour change compared to before is that we're going to pass all the values no matter if they have changed or not (As long as at least one has changed) - why are we changing this behaviour?

Reply:
it's same behavior, we're not really passing all values, only the ones that are set, it will pass nil if the string is empty si it won't go as now

Reply:
From https://github.com/mongodb/terraform-provider-mongodbatlas/pull/2031 it would seem that regionName is the only property that must be included in the update. For the rest we can include them only if they have mainly for consistency with how we handle other attributes in this same function.

Reply:
if you change the provider name you might want to change other attributes as different providers require different attributes.  i've change it to send the require argument depending on the provider.  It's the same idea as we're doing in Create

Reply:
looping in @BrettFolkins for this just for awareness

Reply:
I'm not familiar enough with best practices in terraform providers to have an opinion about sending the request with only the necessary changes or the full target state, but this version lgtm. Thanks!